engine/src/main/java/org/terasology/rendering/shader/ShaderParametersBase.java

```diff
@@ -43,27 +45,29 @@ public void applyParameters(Material program) {
         program.setFloat("viewingDistance", CoreRegistry.get(Config.class).getRendering().getViewDistance().getChunkDistance().x * 8.0f);
 
         WorldRenderer worldRenderer = CoreRegistry.get(WorldRenderer.class);
-        WorldProvider worldProvider = CoreRegistry.get(WorldProvider.class);
+        BackdropProvider backdropProvider = CoreRegistry.get(BackdropProvider.class);
 
-        if (worldRenderer != null) {
-            program.setFloat("daylight", worldRenderer.getDaylight(), true);
+        if (worldRenderer != null && backdropProvider != null) {
```

Comment:
Can I get a comment on this @immortius ?

And regarding the world renderer... sorry, I can't understand if you were being serious asking about it as I assume you already know. In case you don't, I checked and the world rendered gets disposed of in StateIngame.dispose(). That been said, I do not know if it gets removed from the CoreRegistry. It certainly gets overwritten anew in StateLoading.step().